### DISCUSSION TITLE
讨论：移除超过半年无人问津的最佳实践

### DISCUSSION THREAD

**Discussion by MistEO** [MaaXYZ/MaaFramework#467] [Ideas]
同时满足以下几个条件的最佳实践将被移除

1. 超过 6 个月没有新的提交
2. 超过 6 个月无新 issue 及回复
3. star 数量近期几乎无增长

这种一般都是作者跑路了（

当然如果有某些特别优秀的项目，会酌情保留

**Comment by MistEO** [comment]
另外我有个想法，我们按 star 数量给最佳实践排序一下怎么样（是不是有点太势利了x

**Comment by Windsland52** [comment]
可能需要一个综合评价，star占一定的权重（如果能计算的话。
不过这样指标以及权重的确定又是一个问题，或许不按分数细分，只分ABCD这种大的评价/只挑出单独挑出有代表性的实践放到一块儿。

**Comment by Saratoga-Official** [comment]
> 另外我有个想法，我们按 star 数量给最佳实践排序一下怎么样（是不是有点太势利了x

不是很支持，写的游戏比较冷门的话用户本来就少

**Comment by overflow65537** [comment]
应该分两个榜单
#### 一个针对玩家
- 能在没有人工介入的情况下完整跑完一圈就能上

#### 另一个针对开发者
- 根据使用的技术，比如集成/custom action
- 使用pipeline来完成复杂操作的也可以上
- （也许可以让开发者在写一些注释？ #456

二者好像都没有必要根据star来排行
游戏榜单可能需要根据issue来清理有重大bug并且维护者弃坑的项目

开发者榜单就没什么变动的必要

**Comment by MistEO** [comment]
再单独弄一个文档好了，包括一个大而全的目录，来者不拒，任何项目都可以放进去，也不会清

主页的既然叫“最佳实践”，我觉得还是要有一定限制的

**Comment by MistEO** [comment]
感觉又不太好，如果有一些游戏，主要日常部分常年不更新，所以脚本也不需要更新，同时功能比较简单也很稳定，没有什么 bug（所以没有 issue），这种还是应该留着的吧

怎么区分作者跑路了和软件太稳定不需要更新（

**Comment by BrokenArrow1404** [comment]
> 感觉又不太好，如果有一些游戏，主要日常部分常年不更新，所以脚本也不需要更新，同时功能比较简单也很稳定，没有什么 bug（所以没有 issue），这种还是应该留着的吧
> 
> 怎么区分作者跑路了和软件太稳定不需要更新（

原来我想法是直接discussion区开个投票，后来感觉大部分用户应该是上github下一个脚本就跑了，根本不会一直关注这个repo的，估计开了投票也没人参加

**Comment by Saratoga-Official** [comment]
> 感觉又不太好，如果有一些游戏，主要日常部分常年不更新，所以脚本也不需要更新，同时功能比较简单也很稳定，没有什么 bug（所以没有 issue），这种还是应该留着的吧
> 
> 怎么区分作者跑路了和软件太稳定不需要更新（

如果稳定的话star应该也会涨点吧，不能毫无变化

**Comment by overflow65537** [comment]
> 感觉又不太好，如果有一些游戏，主要日常部分常年不更新，所以脚本也不需要更新，同时功能比较简单也很稳定，没有什么 bug（所以没有 issue），这种还是应该留着的吧
> 
> 怎么区分作者跑路了和软件太稳定不需要更新（

查issue啊,假如有个一周前的issue被标记已经解决,但是commit记录最晚都是半年前
那肯定是鸽了

**Comment by MistEO** [comment]
感觉都太主观了而且不好查，还得跟侦探似的（

得找一个比较好量化的指标

**Comment by Saratoga-Official** [comment]
要不让最佳实践里面的作者都加下开发群，如果满足条件了就@一下作者，如果没反应就删了吧

**Comment by MistEO** [comment]
好像可以直接去发个 issue 问一下

**Comment by MistEO** [comment]
最终结论：

同时满足以下几个条件

- 超过 6 个月没有新的提交
- 超过 6 个月无新 issue 及回复

我会去对应项目提一个 issue 询问作者是否弃坑，并告知将从最佳实践移除。

**Reply by MistEO** [reply]
https://github.com/MaaXYZ/MaaResonance/issues/6

**Reply by MistEO** [reply]
https://github.com/MaaXYZ/MBA/issues/15

**Reply by MistEO** [reply]
https://github.com/MaaXYZ/MaaFramework/pull/469

**Comment by dongwlin** [comment]
还可以再开一个used by列表

**Comment by Saratoga-Official** [comment]
还是建议搞个不移除的列表，里面放所有用FW开发的实践，这样有人问还有没有别的游戏的maa的时候我就可以把列表甩给他了

**Reply by Windsland52** [reply]
这就事used by吧

**Reply by Saratoga-Official** [reply]
> 这就事used by吧

我看删完了没加used by列表就再提醒一下

**Comment by MistEO** [comment]
后续在 https://github.com/MaaXYZ/MaaFramework/issues/872 中进行更新
